tests/Mocks/StorageEngineMock.cpp

```diff
@@ -1569,7 +1569,9 @@ int TransactionCollectionMock::use(int nestingLevel) {
     }
   }
 
-  _collection = _transaction->vocbase().useCollection(_cid, status);
+  if (!_collection) {
```

Comment:
how exactly does not incrementing the ref counter when asked to increment the ref counter address deadlocks?